public/app/features/scenes/components/layout/SceneGridLayout.tsx

```diff
@@ -0,0 +1,403 @@
+import { css, cx } from '@emotion/css';
+import React, { useMemo } from 'react';
+import ReactGridLayout from 'react-grid-layout';
+import AutoSizer from 'react-virtualized-auto-sizer';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Icon, useStyles2, useTheme2 } from '@grafana/ui';
+import { DEFAULT_ROW_HEIGHT, GRID_CELL_HEIGHT, GRID_CELL_VMARGIN, GRID_COLUMN_COUNT } from 'app/core/constants';
+
+import { SceneObjectBase } from '../../core/SceneObjectBase';
+import {
+  SceneComponentProps,
+  SceneLayoutChild,
+  SceneLayoutChildState,
+  SceneLayoutState,
+  SceneObject,
+} from '../../core/types';
+import { SceneDragHandle } from '../SceneDragHandle';
+
+interface SceneGridLayoutState extends SceneLayoutState {}
+
+type GridCellLayout = {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+};
+
+export class SceneGridLayout extends SceneObjectBase<SceneGridLayoutState> {
+  static Component = SceneGridLayoutRenderer;
+
+  constructor(state: SceneGridLayoutState) {
+    super({
+      isDraggable: true,
+      ...state,
+    });
+  }
+
+  updateLayout() {
+    this.setState({
+      children: [...this.state.children],
+    });
+  }
+
+  onResizeStop: ReactGridLayout.ItemCallback = (_, o, n) => {
+    const child = this.state.children.find((c) => c.state.key === n.i);
+    if (!child) {
+      return;
+    }
+    child.setState({
+      size: {
+        ...child.state.size,
+        width: n.w,
+        height: n.h,
+      },
+    });
+  };
+
+  // When layout changes, figure out if it's a nested grid layout and update enclosing cell's size if needed
+  onLayoutChange = (layout: ReactGridLayout.Layout[]) => {
+    // debugger;
+    // let enclosingLayout = this.parent ? this.parent.getLayout() : undefined;
+    // if (!enclosingLayout) {
```

Comment:
@torkelo Commented out this for now - I think I will try to solve this in a separate PR as after removing the grid item components, it's not so obvious how nested grid elements should affect other grids.